storage/backup/backup-cli/src/backup_types/epoch_ending/restore.rs

```diff
@@ -43,6 +43,20 @@ impl EpochEndingRestoreController {
     }
 
     pub async fn run(self) -> Result<()> {
+        println!(
+            "Epoch ending restore started. Manifest: {}",
+            self.manifest_handle
+        );
+        self.run_impl()
+            .await
+            .map_err(|e| anyhow!("Epoch ending restore failed: {}", e))?;
+        println!("Epoch ending restore succeeded.");
+        Ok(())
+    }
+}
+
+impl EpochEndingRestoreController {
```

Comment:
does this need to be a separate `impl` block?

Reply:
Most structs in this crate (whereever I remembered), I extracted the pub interfaces (usually few) to a separate impl...